File: FitBook.Model/Responses/Specializations/SpecializationResponse.cs
namespace FitBook.Model.Responses.Specializations;

public class SpecializationResponse : IEntityResponse
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public bool IsActive { get; set; }
    public DateTime CreatedAtUtc { get; set; }
    public DateTime? UpdatedAtUtc { get; set; }
}

File: UI/fitbook_desktop/lib/models/responses/specialization_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'specialization_response.g.dart';

@JsonSerializable()
class SpecializationResponse {
  SpecializationResponse({
    required this.id,
    required this.name,
    required this.isActive,
    required this.createdAtUtc,
    this.updatedAtUtc,
  });

  final int id;
  final String name;
  final bool isActive;
  final DateTime createdAtUtc;
  final DateTime? updatedAtUtc;

  factory SpecializationResponse.fromJson(Map<String, dynamic> json) =>
      _$SpecializationResponseFromJson(json);

  Map<String, dynamic> toJson() => _$SpecializationResponseToJson(this);
}
